pkg/staticnetworkconfig/generator.go

```diff
@@ -148,6 +145,27 @@ func (s *StaticNetworkConfigGenerator) formatNMConnection(nmConnection string) (
 	return buf.String(), nil
 }
 
+func (s *StaticNetworkConfigGenerator) validateAndCalculateMAC(ifaceName string) (string, error) {
+	// check the format of the interface name
+	match, err := regexp.MatchString("^[0-9A-Fa-f]{12}$", ifaceName)
```

Comment:
consider pulling it into a var and use it in error message

Reply:
done